http2_general/test_max_queued_control_frames.py

```diff
@@ -0,0 +1,152 @@
+"""The functional tests for `max_queued_control_frames` directive."""
+
+__author__ = "Tempesta Technologies, Inc."
+__copyright__ = "Copyright (C) 2024 Tempesta Technologies, Inc."
+__license__ = "GPL2"
+
+import socket
+
+from h2.settings import SettingCodes
+from hyperframe.frame import HeadersFrame, PingFrame, PriorityFrame, SettingsFrame
+
+from framework import tester
+from framework.parameterize import param, parameterize
+from helpers import dmesg
+
+
+class TestH2ControlFramesFlood(tester.TempestaTest):
+    clients = [
+        {
+            "id": "deproxy",
+            "type": "deproxy_h2",  # "deproxy" for HTTP/1
+            "addr": "${tempesta_ip}",
+            "port": "443",
+            "ssl": True,
+            "ssl_hostname": "tempesta-tech.com",
+        },
+    ]
+
+    backends = [
+        {
+            "id": "deproxy",
+            "type": "deproxy",
+            "port": "8000",
+            "response": "static",
+            "response_content": (
+                "HTTP/1.1 200 OK\r\n" + "Content-Length: 200000\r\n\r\n" + ("a" * 200000)
+            ),
+        }
+    ]
+
+    tempesta = {
+        "config": """
+        keepalive_timeout 1000;
+        listen 443 proto=h2;
+
+        tls_match_any_server_name;
+
+        srv_group default {
+            server ${server_ip}:8000;
+        }
+        %s
+        vhost tempesta-tech.com {
+           tls_certificate ${tempesta_workdir}/tempesta.crt;
+           tls_certificate_key ${tempesta_workdir}/tempesta.key;
+           proxy_pass default;
+        }
+        """
+    }
+
+    def __update_tempesta_config(self, limit: int) -> None:
+        limit = "" if limit == 10000 else f"max_queued_control_frames {limit};"
+        self.get_tempesta().config.defconfig = self.get_tempesta().config.defconfig % limit
+
+    def __init_connection_and_make_request(self, client) -> None:
+        client.update_initial_settings()
+        client.send_bytes(client.h2_connection.data_to_send())
+        self.assertTrue(client.wait_for_ack_settings())
+
+        client.make_request(client.create_request(method="GET", headers=[]))
```

Comment:
There is no need to request a large response now, because the sockbuf is already set to a small enough size. Also we don't need the backend, because control-frame-slow-read does not involve the backend.